uiowa/uiowa
# [ Registrar ] Course deadlines query parameters.

Closes #8157 


## To test
```
ddev blt ds --site=registrar.uiowa.edu && ddev drush @registrar.local uli /calendars-and-deadlines/course-deadlines
```

1. add `?session=20243` to the URL and enter. Observe that the session field is filled.
2. Additionally, without removing the previous params, add `&department=NURS` to the URL and enter. Observe that the department field is filled.
3. Additionally, without removing the previous params, add `&course=3600` to the URL and enter. Observe that the course field is filled.
4. Additionally, without removing the previous params, add `&section=1018283` to the URL and enter. Observe that the section field is filled.
5. Now that you have one URL structure working, can you mess up what I've built?
6. Test other sessions and sections by getting their keys from the html select element for the session select field. Test the rest of the fields by using their indices, which are the same as their field values.
7. Try and find other bugs?

## To test part 2

1. Ensure that both the deadlines form and the correspondence form are not broken, and can both still use query parameters.
    - https://registrar.uiowa.ddev.site/correspondence
    - https://registrar.uiowa.ddev.site/calendars-and-deadlines/course-deadlines

### NOTE:
- The form is not going to change URL params when you change the form fields.
- If you put in incorrect data into the URL params, it may fill out the fields out of order and disable them incorrectly. This is not something we currently need to worry about. You may disregard.
